src/Tools/dotnet-monitor/Exceptions/ExceptionsService.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.WebApi.Exceptions;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Exceptions
+{
+    /// <summary>
+    /// Get exception information from default process and store it.
+    /// </summary>
+    internal sealed class ExceptionsService :
+        BackgroundService
+    {
+        private readonly TimeSpan RetryDelay = TimeSpan.FromSeconds(5);
+
+        private readonly IExceptionsStore _exceptionsStore;
+        private readonly IDiagnosticServices _diagnosticServices;
+        private readonly IInProcessFeatures _inProcessFeatures;
+
+        private EventExceptionsPipeline _pipeline;
+
+        public ExceptionsService(
+            IDiagnosticServices diagnosticServices,
+            IInProcessFeatures inProcessFeatures,
+            IExceptionsStore exceptionsStore)
+        {
+            _diagnosticServices = diagnosticServices;
+            _exceptionsStore = exceptionsStore;
+            _inProcessFeatures = inProcessFeatures;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            if (!_inProcessFeatures.IsExceptionsEnabled)
+            {
+                return;
+            }
+
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                try
+                {
+                    // Get default process
+                    IProcessInfo pi = await _diagnosticServices.GetProcessAsync(processKey: null, stoppingToken);
+                    DiagnosticsClient client = new(pi.EndpointInfo.Endpoint);
+
+                    EventExceptionsPipelineSettings settings = new();
+                    _pipeline = new EventExceptionsPipeline(client, settings, _exceptionsStore);
+
+                    // Monitor for exceptions
+                    await _pipeline.RunAsync(stoppingToken);
+                }
+                catch (Exception e) when (e is not OperationCanceledException || !stoppingToken.IsCancellationRequested)
```

Comment:
The token is only signaled when the application is being torn down. Background services can either exit gracefully or throw under this circumstance; see https://github.com/dotnet/runtime/blob/210a7a9feec64b7fac5147656cddb199ec90cf75/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L102

This is the same behavior as the metrics service: https://github.com/dotnet/dotnet-monitor/blob/fdbfcaf49ccb104026806dc371ebb63c02d74d49/src/Microsoft.Diagnostics.Monitoring.WebApi/Metrics/MetricsService.cs#L67

I think this was done to avoid waiting 5 seconds on an exception when trying to shutdown. But we can fix both of them to catch but not wait under this circumstance (preferably in a separate PR).